### PR TITLE
aws2-ses: Add support for CC and BCC addresses

### PR DESCRIPTION
**Pull request by chrysophylax**
Implements [camel-17610](https://issues.apache.org/jira/browse/CAMEL-17610)

Adds support for CC and BCC type of destinations. Updates tests too.

### CODE REVIEW

File: components/camel-aws/camel-aws2-ses/src/main/java/org/apache/camel/component/aws2/ses/Ses2Producer.java [hunk 1]
```diff
@@ -148,15 +148,39 @@ private String determineReturnPath(Exchange exchange) {
...
+    private Destination determineDestination(Exchange exchange) {
+        List<String> to = determineRawTo(exchange);
+        List<String> cc = determineRawCc(exchange);
+        List<String> bcc = determineRawBcc(exchange);
+        return Destination.builder().toAddresses(to).ccAddresses(cc).bccAddresses(bcc).build();
+    }
+
+    private List<String> determineRawCc(Exchange exchange) {
+        String cc = exchange.getIn().getHeader(Ses2Constants.CC, String.class);
+        if (cc == null) {
```
Comment (oscerd) [inline components/camel-aws/camel-aws2-ses/src/main/java/org/apache/camel/component/aws2/ses/Ses2Producer.java:160]: Please use ObjectHelper to check for null

File: components/camel-aws/camel-aws2-ses/src/main/java/org/apache/camel/component/aws2/ses/Ses2Producer.java [hunk 2]
```diff
@@ -148,15 +148,39 @@ private String determineReturnPath(Exchange exchange) {
...
+                    .map(String::trim)
+                    .collect(Collectors.toList());
+        } else {
+            return Collections.emptyList();
+        }
+    }
+
+    private List<String> determineRawBcc(Exchange exchange) {
+        String bcc = exchange.getIn().getHeader(Ses2Constants.BCC, String.class);
+        if (bcc == null) {
```
Comment (oscerd) [inline components/camel-aws/camel-aws2-ses/src/main/java/org/apache/camel/component/aws2/ses/Ses2Producer.java:174]: same

**Review by oscerd — CHANGES_REQUESTED** [review]
Just minor stuff.

**Review by oscerd — APPROVED** [review]

**Comment by chrysophylax** [comment]
Hmm, the codeQL check seems to have died halfway because of a build error. Can it be re-run? Build seems to succeed locally and also in the checkstyle build.

**Pull request merged**